docs/usage/models.md

```diff
@@ -643,3 +644,11 @@ In addition, the `**data` argument will always be present in the signature if `C
     Types in the model signature are the same as declared in model annotations, 
     not necessarily all the types that can actually be provided to that field.
     This may be fixed one day once [#1055](https://github.com/pydantic/pydantic/issues/1055) is solved.
+
+## Structural pattern matching
+
+*pydantic* supports structural pattern matching for models, as introduced by [PEP 634](https://peps.python.org/pep-0622/) in Python 3.10.
+
+```py
+{!.tmp_examples/models_structural_pattern_matching.py!}
+```
```

Comment:
can we add a [note admonition](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#admonition-icons) explaining that a new model is not created, and this is just syntactic super form matching?

Dumb or ill-informed people (like me) who haven't used pattern matching before could be confused and worry about a performance overhead for creating models which are not actually created.